### ISSUE TITLE
Building from sources on FreeBSD - "unknown option '-O2'"

### ISSUE THREAD

**Issue by szczepanpp** [HaxeFoundation/haxe#6146]
Hi,

I'm trying to build Haxe compiler from sources on FreeBSD system. I think I have all necessary libs (as listed on https://haxe.org/documentation/introduction/building-haxe.html).

The log output:
```
[(...)]:<~/haxe/git/haxe>$ gmake
make -C libs/extlib native && make -C libs/extc native && make -C libs/neko native && make -C libs/javalib native && make -C libs/ziplib native && make -C libs/swflib native && make -C libs/xml-light native && make -C libs/ttflib native && make -C libs/ilib native && make -C libs/objsize native && make -C libs/pcre native && true
ocamlopt -g -a -o extlib.cmxa enum.mli bitSet.mli dynArray.mli extArray.mli extHashtbl.mli extList.mli extString.mli global.mli rbuffer.mli IO.mli option.mli pMap.mli std.mli uChar.mli uTF8.mli base64.mli unzip.mli refList.mli optParse.mli dllist.mli multiArray.mli sMap.mli enum.ml bitSet.ml dynArray.ml extArray.ml extHashtbl.ml extList.ml extString.ml global.ml rbuffer.ml IO.ml option.ml pMap.ml std.ml uChar.ml uTF8.ml base64.ml unzip.ml refList.ml optParse.ml dllist.ml multiArray.ml sMap.ml extLib.ml
File "bitSet.ml", line 23, characters 40-53:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "bitSet.ml", line 28, characters 64-75:
Warning 3: deprecated: String.fill
Use Bytes.fill instead.
File "extString.ml", line 28, characters 9-15:
Warning 3: deprecated: create
Use Bytes.create instead.
File "extString.ml", line 30, characters 2-12:
Warning 3: deprecated: unsafe_set
File "extString.ml", line 111, characters 2-8:
Warning 3: deprecated: create
Use Bytes.create instead.
File "extString.ml", line 158, characters 9-15:
Warning 3: deprecated: create
Use Bytes.create instead.
File "extString.ml", line 160, characters 21-31:
Warning 3: deprecated: unsafe_set
File "extString.ml", line 165, characters 10-16:
Warning 3: deprecated: create
Use Bytes.create instead.
File "extString.ml", line 167, characters 2-12:
Warning 3: deprecated: unsafe_set
File "extString.ml", line 197, characters 12-25:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "extString.ml", line 200, characters 14-26:
Warning 3: deprecated: String.set
Use Bytes.set instead.
File "extString.ml", line 216, characters 12-18:
Warning 3: deprecated: create
Use Bytes.create instead.
File "extString.ml", line 235, characters 28-39:
Warning 3: deprecated: String.copy
File "rbuffer.ml", line 44, characters 10-23:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "rbuffer.ml", line 71, characters 10-23:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "rbuffer.ml", line 99, characters 10-23:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "rbuffer.ml", line 123, characters 18-31:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "rbuffer.ml", line 135, characters 2-21:
Warning 3: deprecated: String.set
Use Bytes.set instead.
File "rbuffer.ml", line 150, characters 14-27:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "rbuffer.ml", line 157, characters 14-27:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "IO.ml", line 65, characters 9-22:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "IO.ml", line 119, characters 9-22:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "IO.ml", line 176, characters 13-26:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "IO.ml", line 320, characters 6-23:
Warning 3: deprecated: String.unsafe_set
File "IO.ml", line 736, characters 12-25:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "IO.ml", line 753, characters 12-25:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "IO.ml", line 755, characters 2-19:
Warning 3: deprecated: String.unsafe_set
File "IO.ml", line 772, characters 4-21:
Warning 3: deprecated: String.unsafe_set
File "std.ml", line 52, characters 16-29:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "std.ml", line 65, characters 37-50:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "std.ml", line 67, characters 13-26:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "std.ml", line 162, characters 14-27:
Warning 3: deprecated: Obj.final_tag
Replaced by custom_tag.
File "base64.ml", line 98, characters 4-21:
Warning 3: deprecated: String.unsafe_set
File "unzip.ml", line 172, characters 12-25:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "unzip.ml", line 179, characters 9-22:
Warning 3: deprecated: String.create
Use Bytes.create instead.
File "unzip.ml", line 191, characters 1-18:
Warning 3: deprecated: String.unsafe_set
File "unzip.ml", line 259, characters 1-18:
Warning 3: deprecated: String.unsafe_set
File "unzip.ml", line 436, characters 9-22:
Warning 3: deprecated: String.create
Use Bytes.create instead.
ocamlc -O2 -pipe   extc_stubs.c
/usr/local/bin/ocamlc: unknown option '-O2'.
Usage: ocamlc <options> <files>
Options are:
  -a  Build a library
  -absname  Show absolute filenames in error messages
  -annot  Save information in <filename>.annot
  -bin-annot  Save typedtree in <filename>.cmt
  -c  Compile only (do not link)
  -cc <command>  Use <command> as the C compiler and linker
  -cclib <opt>  Pass option <opt> to the C linker
  -ccopt <opt>  Pass option <opt> to the C compiler and linker
  -compat-32  Check that generated bytecode can run on 32-bit platforms
  -config  Print configuration values and exit
  -custom  Link in custom mode
  -custom  Link in custom mode
  -dllib <lib>  Use the dynamically-loaded library <lib>
  -dllpath <dir>  Add <dir> to the run-time search path for shared libraries
  -dtypes  (deprecated) same as -annot
  -for-pack <ident>  Generate code that can later be `packed' with
     ocamlc -pack -o <ident>.cmo
  -g  Save debugging information
  -i  Print inferred interface
  -I <dir>  Add <dir> to the list of include directories
  -impl <file>  Compile <file> as a .ml file
  -intf <file>  Compile <file> as a .mli file
  -intf-suffix <string>  Suffix for interface files (default: .mli)
  -intf_suffix <string>  (deprecated) same as -intf-suffix
  -keep-docs  Keep documentation strings in .cmi files
  -keep-locs  Keep locations in .cmi files
  -labels  Use commuting label mode
  -linkall  Link all modules, even unused ones
  -make-runtime  Build a runtime system with given C objects and libraries
  -make_runtime  (deprecated) same as -make-runtime
  -modern  (deprecated) same as -labels
  -no-alias-deps  Do not record dependencies for module aliases
  -no-app-funct  Deactivate applicative functors
  -no-check-prims  Do not check runtime for primitives
  -noassert  Do not compile assertion checks
  -noautolink  Do not automatically link C libraries specified in .cma files
  -nolabels  Ignore non-optional labels in types
  -nostdlib  Do not add default directory to the list of include directories
  -o <file>  Set output file name to <file>
  -open <module>  Opens the module <module> before typing
  -output-obj  Output an object file instead of an executable
  -output-complete-obj  Output an object file, including runtime, instead of an executable
  -pack  Package the given .cmo files into one .cmo
  -pp <command>  Pipe sources through preprocessor <command>
  -ppx <command>  Pipe abstract syntax trees through preprocessor <command>
  -principal  Check principality of type inference
  -rectypes  Allow arbitrary recursive types
  -runtime-variant <str>  Use the <str> variant of the run-time system
  -safe-string  Make strings immutable
  -short-paths  Shorten paths in types
  -strict-sequence  Left-hand part of a sequence must have type unit
  -strict-formats  Reject invalid formats accepted by legacy implementations
     (Warning: Invalid formats may behave differently from
      previous OCaml versions, and will become always-rejected
      in future OCaml versions. You should use this flag
      to detect and fix invalid formats.)
  -thread  Generate code that supports the system threads library
  -unsafe  Do not compile bounds checking on array and string access
  -unsafe-string  Make strings mutable (default)
  -use-runtime <file>  Generate bytecode for the given runtime system
  -use_runtime <file>  (deprecated) same as -use-runtime
  -v  Print compiler version and location of standard library and exit
  -verbose  Print calls to external commands
  -version  Print version and exit
  -vmthread  Generate code that supports the threads library with VM-level
     scheduling
  -vnum  Print version number and exit
  -w <list>  Enable or disable warnings according to <list>:
        +<spec>   enable warnings in <spec>
        -<spec>   disable warnings in <spec>
        @<spec>   enable warnings in <spec> and treat them as errors
     <spec> can be:
        <num>             a single warning number
        <num1>..<num2>    a range of consecutive warning numbers
        <letter>          a predefined set
     default setting is "+a-4-6-7-9-27-29-32..39-41..42-44-45-48-50"
  -warn-error <list>  Enable or disable error status for warnings according
     to <list>.  See option -w for the syntax of <list>.
     Default setting is "-a"
  -warn-help  Show description of warning numbers
  -where  Print location of standard library and exit
  - <file>  Treat <file> as a file name (even if it starts with `-')
  -nopervasives  (undocumented)
  -use-prims <file>  (undocumented)
  -dsource  (undocumented)
  -dparsetree  (undocumented)
  -dtypedtree  (undocumented)
  -drawlambda  (undocumented)
  -dlambda  (undocumented)
  -dinstr  (undocumented)
  -help  Display this list of options
  --help  Display this list of options
*** Error code 2

Stop.
make[1]: stopped in /usr/home/(...)/haxe/git/haxe/libs/extc
gmake: *** [Makefile:100: libs] Błąd 1
```

Versions:
Haxe GIT HEAD commit.
OCaml version 4.02.3
FreeBSD 10.3-RELEASE-p17.

I'm able to build Haxe on Ubuntu without any problems but I'm completely new to FreeBSD, so excuse me if I'm missing something obvious.

Thanks!

**Comment by andyli** [comment]
What exact Haxe commit you're compiling?
It sounds like related to the problem I fixed in https://github.com/HaxeFoundation/haxe/commit/03598f035c40928af7be610b58108fd0227a0869

**Comment by szczepanpp** [comment]
Did fetch/pull and now I'm at current HEAD commit, which would be dea00edac86f31ab492b86475f168c7e7e6ac87d. I can see in the log those changes you mentioned, still it fails to compile.

**Comment by andyli** [comment]
I've committed a fix to the Makefiles in c8e43d38c53638614bb5348ad17d33e16489867f, but I haven't checked fully since I don't know how to install development headers (e.g. pcre.h) in FreeBSD (How to do that?).

**Comment by andyli** [comment]
Ahh, the pcre.h is already shipped by the pcre package and installed to `/usr/local/include`. I can compile haxe by the command as follows:

```bash
gmake "CFLAGS=-I /usr/local/include/"
```

For reference, I installed `ocaml`, `ocaml-findlib`, `ocaml-camp4`, `pcre`, `gmake`.

I think it works pretty nicely on FreeBSD now, so I'm going to close this. Let me know if there is any other problem.
